chain/network/tests/peer_handshake.rs

```diff
@@ -209,5 +209,12 @@ fn check_connection_with_new_identity() {
     runner.push(Action::CheckRoutingTable(0, vec![(1, vec![1])]));
     runner.push(Action::CheckRoutingTable(1, vec![(0, vec![0])]));
 
+    runner.push(Action::Wait(2000));
+
+    // Check the no node tried to connect to itself in this process.
+    runner.push_action(wait_for(|| {
```

Comment:
Is there a better way to do this? I am not a fan of adding a counter just for testing purposes.

Reply:
This counter is indeed somehow "useful", since it is still possible you try to connect to yourself in very weird situations (more likely because you have some complex network setup and reset your node with a new PeerID).

Other approach to test this are quite involved. For example we need to store an integer/boolean in state to know whether we tried to request info to ourself, and create Actix Message Response to query PeerManagerActor for this information. 

I think it will introduce some extra unnecessary complexity. I'm not fan either of using counters for testing, but in this case, I think it is not a bad idea. One option is to hide that counter behind a flag, but I still don't think we should go that far for something this simple.

Reply:
Okay I cannot think of a better way either.